test/gtest/lstm_extra.cpp

```diff
@@ -0,0 +1,204 @@
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-cx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-hx --no-cx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-dcy"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-cx --no-dcy"},
+        std::pair{env, flags + commonFlags + " -dir-mode 0 --no-hy"},
+        std::pair{env, flags + commonFlags + " -dir-mode 0 --no-dhx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 0 --no-hy --no-dhx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 0 --no-cy"},
+        std::pair{env, flags + commonFlags + " -dir-mode 0 --no-hy --no-cy"},
+        std::pair{env, flags + commonFlags + " -dir-mode 0 --no-dcx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 0 --no-cy --no-dcx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-hy"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-dhx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-hy --no-dhx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-cy"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-hy --no-cy"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-dcx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-cy --no-dcx"},
+        std::pair{
+            env,
+            flags + commonFlags +
+                " -dir-mode 0 --no-hx --no-dhy --no-cx --no-dcy --no-hy --no-dhx --no-cy --no-dcx"},
+        std::pair{env,
+                  flags + commonFlags +
+                      " -dir-mode 1 --no-hx --no-dhy --no-cx --no-dcy --no-hy --no-dhx --no-cy "
+                      "--no-dcx"}};
+}
+
+using TestCase = decltype(GetTestCases({}))::value_type;
+
+class ConfigWithFloat : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+class ConfigWithHalf : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+bool IsTestSupportedForDevice()
+{
+    using namespace miopen::debug;
+    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx103X, Gpu::gfx110X>;
+    using d_mask = disabled<Gpu::gfx900, Gpu::gfx906, Gpu::gfx908, Gpu::gfx90A>;
+    return miopen::debug::IsTestSupportedForDevice<d_mask, e_mask>();
+}
+
+void Run2dDriver(miopenDataType_t prec)
+{
+    if(!IsTestSupportedForDevice())
+    {
+        GTEST_SKIP();
+    }
+    std::vector<std::pair<EnvType, std::string>> params;
+    switch(prec)
+    {
+    case miopenFloat: params = ConfigWithFloat::GetParam(); break;
+    case miopenHalf: params = ConfigWithHalf::GetParam(); break;
+    case miopenFloat8:
+    case miopenBFloat8:
+    case miopenInt8:
+    case miopenBFloat16:
+    case miopenInt32:
+    case miopenDouble:
+        FAIL() << "miopenInt8, miopenBFloat16, miopenInt32, "
+                  "miopenDouble, miopenFloat8, miopenBFloat8 "
+                  "data types not supported by "
+                  "lstm_extra test";
+
```

Comment:
No needed.
```suggestion
```

Reply:
@atamazov , just to clarify this function looks redundant to me since the env var wouldn't be needed. But the above mentioned blurb should stay except for

https://github.com/ROCm/MIOpen/blob/3a32d798353462670c003ae0d68fbd78261b4874/test/gtest/lstm_extra.cpp#L39C1-L55C2

But the above mentioned line should stay to get "params" except for this line since half precision is removed, right? Thanks

https://github.com/ROCm/MIOpen/blob/3a32d798353462670c003ae0d68fbd78261b4874/test/gtest/lstm_extra.cpp#L140

Reply:
[resolved]